record/program-anchor/programs/record-anchor/src/lib.rs

```diff
@@ -0,0 +1,150 @@
+        }
+
+        account_data.authority = authority_info;
+        account_data.version = CURRENT_VERSION;
+        account_data.data.bytes = [111u8; DATA_SIZE];
+
+        Ok(())
+    }
+
+    pub fn set_authority(ctx: Context<SetAuthority>) -> Result<()> {
+        msg!("RecordInstruction::SetAuthority");
+        let data_info = &mut ctx.accounts.record_account;
+        let new_authority_info = ctx.accounts.new_authority.key();
+        let mut account_data = RecordData::try_from_slice(&data_info.data.bytes)?;
+
+        if !account_data.is_initialized() {
+            msg!("Record account not initialized");
+            return Err(ProgramError::UninitializedAccount.into());
+        }
+        account_data.authority = new_authority_info;
+        Ok(())
+    }
+
+    pub fn close_account(ctx: Context<CloseAccount>) -> Result<()> {
+        msg!("RecordInstruction::CloseAccount");
+        let data_info = &mut ctx.accounts.record_account;
+        let destination_info = &mut ctx.accounts.reciever;
+        let account_data = &mut RecordData::try_from_slice(&data_info.data.bytes)?;
+        if !account_data.is_initialized() {
+            msg!("Record not initialized");
+            return Err(ProgramError::UninitializedAccount.into());
+        }
+
+        let destination_starting_lamports = destination_info.lamports();
+        let data_lamports = data_info.to_account_info().lamports();
+        
+        **data_info.to_account_info().lamports.borrow_mut() = 0;
+        **destination_info.to_account_info().lamports.borrow_mut() = destination_starting_lamports
+            .checked_add(data_lamports)
+            .ok_or(ProgramError::Custom(0))?;
+        account_data.data = Data::default();
+        Ok(())
+    }
+
+    pub fn write(ctx: Context<Write>, offset: u64, data: Vec<u8>) -> Result<()> {
+        msg!("RecordInstruction::Write");
+        let data_info = &mut ctx.accounts.record_account;
+        let account_data = RecordData::try_from_slice(&data_info.data.bytes)?;
+        if !account_data.is_initialized() {
+            msg!("Record account not initialized");
+            return Err(ProgramError::UninitializedAccount.into());
+        }
+        let start = WRITABLE_START_INDEX + offset as usize;
+        let end = start + data.len();
+        if end > data_info.data.bytes.len() {
+            return Err(ProgramError::AccountDataTooSmall.into());
+        } else {
+            data_info.data.bytes[start..end].copy_from_slice(&data);
+        }
+        Ok(())
+    }
+}
+
+#[derive(Accounts)]
+pub struct Initialize<'info> {
+    #[account(init, payer = authority, space = 8 + 32 + 1 + 8)]
+    pub record_account: Account<'info, RecordData>,
+    #[account(mut)]
+    pub authority: Signer<'info>,
+    pub system_program: Program<'info, System>
+}
+
+#[derive(Accounts)]
+pub struct SetAuthority<'info> {
+    #[account(mut, has_one = authority)]
+    pub record_account: Account<'info, RecordData>,
+    #[account(mut)]
```

Comment:
This probably doesn't need to be `mut` since the data and lamports aren't changing

Reply:
Makes sense, done!